### PR TITLE
Ft reset password page 167131479

### PR DESCRIPTION
**Pull request by beejay1293**
### What does this PR do?

* set up reset password page

### Description of Task to be completed?

* setup auth reducer
* create reset password page
* set up reset password component
* set up action with axios to consume endpoint from backend

### Any background context you want to provide?

N/A

### What are the relevant Github issues?

N/A

### Questions:

N/A

### Screenshot:

image

### CODE REVIEW

File: store/actions/authActions/resetPassword.js [hunk 1]
```diff
@@ -28,6 +29,7 @@ export const ResetPassword = user => async (dispatch) => {
       },
     });
 
+    // console.log(resetPassword);
```
Comment (chekwas88) [inline store/actions/authActions/resetPassword.js]: nice job bolaji. Please remove the commented statements

File: src/components/Button/button.scss [hunk 2]
```diff
@@ -1,8 +1,7 @@
 $btn: #f3994a;
 $button: #fff;
-
-.sign-form .b-button {
-  background-color: $btn;
+.b-button {
+  background-color: #f3994a;
```
Comment (fortunee) [inline src/components/Button/button.scss]: Why aren't you using the $btn here?

File: src/components/ChangePassword/ChangePassword.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+import * as actions from '../../../store/actions/authActions/changePassword';
+import Card from '../Card/Card';
+import Button from '../Button';
+import Input from '../Inputs/Input';
+import Footer from '../Footer/Footer';
+import { Header } from '../Header/Header';
+import { Heading } from '../Heading/Heading';
+
+/**
+ * @description
```
Comment (fortunee) [inline src/components/ChangePassword/ChangePassword.js]: The docstring comments here doesn't make sense, please refactor. @beejay1293

File: src/components/ChangePassword/ChangePassword.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    this.setState({ [name]: value });
+  }
+
+  async handleSubmit() {
+    const { password, confirmPassword } = this.state;
+    const { changePassword, history, location } = this.props;
+    this.setState({ passwordFieldError: '' });
+    this.setState({ confirmPasswordFieldError: '' });
+    this.setState({ error: '' });
+    this.setState({ success: '' });
```
Comment (fortunee) [inline src/components/ChangePassword/ChangePassword.js]: Just out of curiosity here, may I know why you couldn't just do this? 👇 

```js
   this.setState({
       passwordFieldError: '',
       confirmPasswordFieldError: '',
       error: '',
       success: ''
   });
```

File: src/components/ChangePassword/ChangePassword.js [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    if (password === '') {
+      this.setState({ passwordFieldError: 'password field cannot be empty' });
+    } else if (password === confirmPassword) {
```
Comment (fortunee) [inline src/components/ChangePassword/ChangePassword.js]: Consider stripping the passwords of any whitespaces @beejay1293

File: src/components/ChangePassword/ChangePassword.js [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+      const { token } = queryString.parse(location.search);
+      await changePassword(password, token, history);
+      const { passwordChangeSuccess, passwordChangeError } = this.props;
+      this.setState({ error: passwordChangeError });
+      this.setState({ password: '', confirmPassword: '' });
+      this.setState({ success: passwordChangeSuccess });
```
Comment (fortunee) [inline src/components/ChangePassword/ChangePassword.js]: I still don't understand why we can't handle all of this in one single `this.setState()`? @beejay1293

File: src/components/ChangePassword/ChangePassword.js [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+                value={confirmPassword}
+                type="password"
+              />
+              <div className="error">{confirmPasswordFieldError}</div>
+            </FormGroup>
+            <Button
+              classname="buttons"
+              type="button"
+              onClick={this.handleSubmit}
+              text={loading === true ? 'Loading...' : 'Change Password'}
```
Comment (fortunee) [inline src/components/ChangePassword/ChangePassword.js]: This works too 

```js
text={loading ? 'Loading...' : 'Change Password'}
```

File: src/components/ResetPassword/resetpassword.scss [hunk 8]
```diff
@@ -72,54 +84,65 @@ $success-color: #008000;
...
+      box-shadow: none;
+      background-color: $form-color !important;
+      border-color: $form-color !important;
+      box-shadow: none !important;
+      opacity: 0.6;
     }
+    &:focus {
+      box-shadow: none;
+      background-color: $form-color !important;
+      border-color: $form-color !important;
```
[the remark this answers is not in the document]
Comment (fortunee) [reply src/components/ResetPassword/resetpassword.scss]: Yes Mr. HoundCI, I don't approve too  ❌ ❌ ❌

File: src/views/Login/Login.js [hunk 9]
```diff
@@ -138,19 +140,21 @@ export class Login extends Component {
...
+              <p>
+                <Link className="forgot-password" to="/password-reset">
+                  Forgot Password?
+                </Link>
+              </p>
               <p>
                 Dont have an account yet?
-                <Link className="sign-up" to="/sign-up"> Sign Up</Link>
+                <Link className="sign-up" to="/sign-up">
+                  {' '}
```
Comment (fortunee) [inline src/views/Login/Login.js]: But brother why? 🤷‍♂

File: src/views/Signup/SignUpForm.js [hunk 10]
```diff
@@ -11,103 +11,105 @@ import Input from '../../components/Inputs/Input';
...
-              {' '}
-            </Link>
-            <br />
-            <div>
-              <Link to="/signup" className="font">
-                <FontAwesomeIcon icon={['fab', 'twitter']} className="icon alt" />
-              </Link>
-              <Link to="/signup" className="font mid">
-                <FontAwesomeIcon icon={['fab', 'facebook-f']} className="icon alt" />
+const SignUpForm = ({ onSubmit, onChange, authError, valError }) => {
```
[the remark this answers is not in the document]
Comment (fortunee) [reply src/views/Signup/SignUpForm.js]: Yes Mr. Hound I support you on this one ooo

File: store/actions/authActions/changePassword.js [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+          accept: 'application/json',
+          'Content-type': 'application/json; charset=UTF-8',
+        },
+      },
+    );
+
+    dispatch(passwordChangeSuccess(resetPassword.data.message));
+  } catch (error) {
+    dispatch(passwordChangeError('please send reset password request again'));
+    // console.log(error.response.data.message);
```
Comment (fortunee) [inline store/actions/authActions/changePassword.js]: Oh no no no no!!!!. Please get rid of this bro, please please.

**Review by fortunee — CHANGES_REQUESTED** [review]
Bolaji, thanks for the good work. However, I left some feedback that needs to be resolved.

**Review by fortunee — APPROVED** [review]
👍

**Pull request merged**